doc/languages-frameworks/maven.section.md

```diff
@@ -49,6 +49,77 @@ This package calls `maven.buildMavenPackage` to do its work. The primary differe
 After setting `maven.buildMavenPackage`, we then do standard Java `.jar` installation by saving the `.jar` to `$out/share/java` and then making a wrapper which allows executing that file; see [](#sec-language-java) for additional generic information about packaging Java applications.
 :::
 
+### Overriding Maven package attributes {#maven-overriding-package-attributes}
+
+The `buildMavenPackage` function has a `overrideMavenAttrs` method that can be
+used to override the attributes of a maven package. In the following example we
+want to use an older version of jd-cli and disable some flaky test:
+
+```nix
+jd-cli.overrideMavenAttrs(old: rec {
+  version = "1.2.0";
+  src = fetchFromGitHub {
+    owner = old.src.owner;
+    repo = old.src.repo;
+    rev = "${old.pname}-${version}";
+    # old source hash of 1.2.0 version
+    hash = "sha256-US7j6tQ6mh1libeHnQdFxPGoxHzbZHqehWSgCYynKx8=";
+  };
+
+  # tests can be disabled by prefixing it with !
+  # see the docs for more info: https://maven.apache.org/surefire/maven-surefire-plugin/examples/single-test.html#Multiple_Formats_in_One
+  mvnParameters = lib.escapeShellArgs [
+    "-Dsurefire.failIfNoSpecifiedTests=false"
+    "-Dtest=!JavaDecompilerTest#basicTest,
+    !JavaDecompilerTest#patternMatchingTest"
+  ];
+
+  # old mvnHash of 1.2.0 maven dependencies
+  mvnHash = "sha256-N9XC1pg6Y4sUiBWIQUf16QSXCuiAPpXEHGlgApviF4I=";
+});
+```
+
+### Offline build {#maven-offline-build}
+
+The default behaviour of the `buildMavenPackage` does the following:
+
+1. Runs `mvn package -Dmaven.repo.local=$out/.m2 ${mvnParameters}` in the
+   `fetchedMavenDeps` fixed output derivation.
+2. Runs `mvn package -o -nsu "-Dmaven.repo.local=$mvnDeps/.m2"
+   ${mvnParameters}` again in the main derivation.
+
+As a result, tests are run twice which also means that a failing test
+triggers a new build of the fixed output derivation which re-downloads all
+dependencies again. For bigger Maven projects, this might lead to a slow
+feedback cycle and wastes build time resources by executing the build twice.
```

Comment:
```suggestion
As a result, tests are run twice.
This also means that a failing test will trigger a new attempt to realise the fixed-output derivation, which in turn downloads all dependencies again.
For bigger Maven projects, this might lead to a long feedback cycle. 
```